### PR TITLE
owner, redo(ticdc): decouple globalResolvedTs and globalBarrierTs

### LINKED ISSUE #8963: decouple globalResolvedTs and globalBarrierTs
In the current changefeed implementation, [globalResolvedTs will be stuck by the globalBarrierTs](https://github.com/pingcap/tiflow/blob/master/cdc/owner/changefeed.go#L380-L384
), which will delay progress of redo resolvedTs.

To solve this problem, we should decouple globalResolvedTs and globalBarrierTs first, and then add a redo ddl barrier to prevent resolvedTs from being advanced too early.

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #8963

### What is changed and how it works?
1. [Decouple globalResolvedTs and globalBarrierTs](https://github.com/pingcap/tiflow/pull/8964/files#diff-953746562776cf8aa41cd4a971df8c546649d2721ae787607aed2069f3b44eb0R381-R382).
2. [Add a redo ddl barrier to prevent resolvedTs from being advanced too early](https://github.com/pingcap/tiflow/pull/8964/files#diff-22d25e82af60f580f09cf3ace50f4c276f10b5e0d29beb6f4999598605bb39f4R457-R465).

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`Fix the isuue that resolvedTs may be advanced too early when redo enabled`.
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

**Review by nongfushanquan — COMMENTED** [review]
could you add some comments in detail about the issue

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -443,8 +443,8 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
 		}
 	})
 
-	c.updateStatus(newCheckpointTs, newResolvedTs, minTableBarrierTs)
-	c.updateMetrics(currentTs, newCheckpointTs, metricsResolvedTs)
+	c.updateStatus(newCheckpointTs, newResolvedTs, barrier.minDDLBarrierTs)
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:446]: which one is the globalBarrierTs now?

Comment (CharlesCheung96) [reply cdc/owner/changefeed.go:446]: `updateStatus` only update status in etcd, however globalBarrierTs is updated by p2p, ref: https://github.com/pingcap/tiflow/pull/8964/files#diff-953746562776cf8aa41cd4a971df8c546649d2721ae787607aed2069f3b44eb0R353-R354

**Comment by CharlesCheung96** [comment]
> could you add some comments in detail about the issue

updated, PTAL: https://github.com/pingcap/tiflow/issues/8963

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**